leftwm-core/src/models/window.rs

```diff
@@ -118,6 +118,10 @@ impl Window {
         self.is_floating = value;
     }
 
+    pub fn explicit_floating(&self) -> bool {
```

Comment:
We need to check if the floating was explicitly enabled via the config as we don't want to call `set_relative_floating` for every floating window. I'm not really satisfied with this because is_floating could be enabled independently of the spawn_floating setting, I could add another property on window for this. What do you think?

Reply:
I am not sure if I follow right through and I might miss some context right now, as I am just reading on mobile.
But can't we just check again if any conditions for setting a window floating are (still) met?
I mean we have it all that in state already, right?

But if you think adding a field to the window might ease things, just go ahead. Might be only a bool anyways, right?

Reply:
The problem is that we don't want to call set_relative_floating with workspace independent windows like polybar. If  window.floating() is used as condition then one polybar will be centered in a multi monitor setup, because all polybars will be added to the active workspace. Is there a util to check if a window is workspace independent?

Reply:
I see.

AFAIK we don't have such a utility. So far most of the time we just check against the window type.
E.g. a polybar is most certainly of type `dock`; or we simply demand to be 😛

But said utility would be certainly a nice addition. Might be even helpful in the longrun, when efforts will be maybe to make the core library independent of x11...

Reply:
Okay, I think we can do something similar here. I changed it to only set_relative_floating if it's a WindowType normal. I think this is sufficient because the other window types mostly have different handling for floating windows. This fixes the problem for my yubioauth thingy and microsoft teams notification (if they both are configured manually in the window_rules)

Reply:
By 'configured manually' you mean just setting the window_rule for floating, right?

I guess eventually something like this might be nice, but can be kept for another PR:
```
  {window_class = "pinentry-gtk-2", spawn_floating = true, x = 10, y = 65},
```

Reply:
Yes, I mean the window_rules. Additionally to the x, y coordinates we should allow setting the width + height inside the window_rules. But yeah, this can be done in a separate PR.

Is there anything else for me to do to make this PR ready to merge?

Reply:
Agreed, `with` and `height` should also be possible, but I think it should be optional which of the four gets configured explicitly. I intended to mention it, but got rushed lunch being ready and forgot :angel:

I think this PR is ready.